package/MDAnalysis/guesser/default_guesser.py

```diff
@@ -0,0 +1,480 @@
+# vim: tabstop=4 expandtab shiftwidth=4 softtabstop=4
+#
+# MDAnalysis --- https://www.mdanalysis.org
+# Copyright (c) 2006-2017 The MDAnalysis Development Team and contributors
+# (see the file AUTHORS for the full list of names)
+#
+# Released under the GNU Public Licence, v2 or any higher version
+#
+# Please cite your use of MDAnalysis in published work:
+#
+# R. J. Gowers, M. Linke, J. Barnoud, T. J. E. Reddy, M. N. Melo, S. L. Seyler,
+# D. L. Dotson, J. Domanski, S. Buchoux, I. M. Kenney, and O. Beckstein.
+# MDAnalysis: A Python package for the rapid analysis of molecular dynamics
+# simulations. In S. Benthall and S. Rostrup editors, Proceedings of the 15th
+# Python in Science Conference, pages 102-109, Austin, TX, 2016. SciPy.
+# doi: 10.25080/majora-629e541a-00e
+#
+# N. Michaud-Agrawal, E. J. Denning, T. B. Woolf, and O. Beckstein.
+# MDAnalysis: A Toolkit for the Analysis of Molecular Dynamics Simulations.
+# J. Comput. Chem. 32 (2011), 2319--2327, doi:10.1002/jcc.21787
+
+from .base import GuesserBase
+import numpy as np
+import warnings
+
+import re
+
+from ..lib import distances
+from . import tables
+
+class DefaultGuesser(GuesserBase):
+    """
+    this guesser hold generic methods (not directed to specific context) for guessing different topology attribute. 
+    it has the same methods that was originally found in Topology.guesser.py. The attributes that can be
+    guessed by this class are:
+       masses 
+       types
+       elements
+       angles
+       dihedrals
+       bonds
+       improper dihedrals
+       aromaticities
+     
+     you can use this guesser either directly by initiating an object of it or through
+     universe.guess_TopologyAttributes API.
+
+     Examples
+     --------
+      to guess bonds for a universe:
+      
+      import MDAnalysis as mda
+      from MDAnalysisTests.datafiles import two_water_gro
+
+      u = mda.Universe(two_water_gro, to_guess=['bonds'])
+      
+    """
+    context = 'default'
+
+    def __init__(self, universe, **kwargs):
+        super().__init__(universe, **kwargs)
+        self._guess = {'masses': self.guess_masses,
+                       'types': self.guess_types,
+                       'elements' : self.guess_types,
+                       'angles': self.guess_angles,
+                       'dihedrals': self.guess_dihedrals,
+                       'bonds': self.guess_bonds,
+                       'improper dihedrals': self.guess_improper_dihedrals,
+                       'aromaticities': self.guess_aromaticities,
+                       }
+
+    def guess_masses(self, atoms=None):
+        """Guess the mass of many atoms based upon their type
+
+        Returns
+        -------
+        atom_masses : np.ndarray dtype float64
+        """
+        atom_types = None
+        parser = self._kwargs.get('parser', [])
+        if atoms is not None:
+            atom_types = atoms
+            
+        elif 'TXYZ' in parser or 'ARC' in parser:
```

Comment:
```suggestion
        elif parser in ('TXYZ', 'ARC'):
```

By inverting the test, you make it easier to read. It is also easier to add parsers to the list would we need to.